tools/jit/gen_unboxing.py

```diff
@@ -0,0 +1,182 @@
+            f, method=(Variant.method in f.variants)
+        )
+
+        sig = sig_group.most_faithful_signature()
+
+        # escape double quote in schema, get rid of extra double quotes
+        schema = json.dumps(sig.func.__str__())[1:-1]
+
+        return f"""
+OperatorGenerator(
+    TORCH_SELECTIVE_SCHEMA("aten::{schema}"),
+    [](Stack & stack) {{
+        RECORD_FUNCTION("{sig.name()}", std::vector<c10::IValue>());
+        at::unboxing::{f.func.name.unambiguous_name()}(stack);
+    }},
+    aliasAnalysisFromSchema()
+),
+"""
+
+
+def gen_unboxing(
+        *,
+        native_functions: Sequence[NativeFunction],
+        cpu_fm: FileManager,
+) -> None:
+    def key_func(fn: Union[NativeFunction, NativeFunctionsGroup]) -> str:
+        return fn.root_name
+
+    cpu_fm.write_sharded(
+        "UnboxingFunctions.cpp",
+        native_functions,
+        key_fn=key_func,
+        env_callable=lambda fn: {
+            "definitions": [ComputeUnboxingFunctions(Target.DEFINITION)(fn)]
+        },
+        num_shards=5,
+        sharded_keys={"definitions"},
+    )
+    cpu_fm.write(
+        "UnboxingFunctions.h",
+        lambda: {
+            "declarations": list(
+                mapMaybe(ComputeUnboxingFunctions(Target.DECLARATION), native_functions)
+            ),
+        },
+    )
+    cpu_fm.write_sharded(
+        "RegisterCodegenUnboxedKernels.cpp",
+        native_functions,
+        key_fn=key_func,
+        env_callable=lambda fn: {"unboxed_ops": [ComputeCodegenUnboxedKernels()(fn)]},
+        num_shards=5,
+        sharded_keys={"unboxed_ops"},
+    )
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Generate unboxing source files")
+    parser.add_argument(
+        "-s",
+        "--source-path",
+        help="path to source directory for ATen",
+        default="aten/src/ATen",
+    )
+    parser.add_argument(
+        "-d", "--install_dir", help="output directory", default="build/aten/src/ATen"
+    )
+    parser.add_argument(
+        '-o',
+        '--output-dependencies',
+        help='output a list of dependencies into the given file and exit')
+    parser.add_argument(
+        '--dry-run', action='store_true',
+        help='run without writing any files (still updates outputs)')
+
+    options = parser.parse_args()
+
+    native_yaml_path = os.path.join(options.source_path, "native/native_functions.yaml")
+    parsed_yaml = parse_native_yaml(native_yaml_path)
+    native_functions, backend_indices = (
+        parsed_yaml.native_functions,
+        parsed_yaml.backend_indices,
+    )
+
+    cpu_fm = make_file_manager(options=options)
+    gen_unboxing(native_functions=native_functions, cpu_fm=cpu_fm)
+
+    if options.output_dependencies:
+        depfile_path = pathlib.Path(options.output_dependencies).resolve()
+        depfile_name = depfile_path.name
+        depfile_stem = depfile_path.stem
+
+        path = depfile_path.parent / depfile_name
```

Comment:
@peterbell10 Do you think there's some refactoring we could have done to reduce the amount of boilerplate code here? This seems like something fm should be able to deal with...

Reply:
I don't think so because this code is over-complicated. It's equivalent to:
```python
    if options.output_dependencies:
        depfile_path = pathlib.Path(options.output_dependencies).resolve()
        fm.write_outputs(depfile_path.stem, str(depfile_path))

```
`.resolve()` is optional. `.stem` is the file name minus file extension, and is used as the cmake variable's name.

The over-complication comes from copy-pasting from `gen.py` which is more complicated because it's generating multiple different filenames from one path. But that isn't happing here.